OnPod/Node/Common/PeripheralInterfaces/include/I2CInterface.h

```diff
@@ -0,0 +1,53 @@
+#ifndef H_I2CINTERFACE
+#define H_I2CINTERFACE
+
+#include "main.h"
+
+class I2CInterface
+{
+    private:
+
+    // Pointer to I2C Handle Type Def structure contains the config info for the specified I2C.
+    I2C_HandleTypeDef *m_hi2c;
+    uint8_t m_dataRecieved[8]; 
+    uint8_t m_dataBufferLength = 8;
```

Comment:
Should we do something about this 8? It kinda seems like a magic number but idk how else to do it

Reply:
Other than std::vector or dynamically allocating the memory I can't see a way to dynamically size the buffer.

Reply:
Should be good for now